sentry/src/main/java/io/sentry/cache/EnvelopeCache.java

```diff
@@ -122,13 +122,17 @@ private boolean storeInternal(final @NotNull SentryEnvelope envelope, final @Not
     if (HintUtils.hasType(hint, AbnormalExit.class)) {
       tryEndPreviousSession(hint);
     }
+    // TODO: adapt       tryEndPreviousSession(hint); to CrashExit.class
 
     if (HintUtils.hasType(hint, SessionStart.class)) {
       movePreviousSession(currentSessionFile, previousSessionFile);
       updateCurrentSession(currentSessionFile, envelope);
 
       boolean crashedLastRun = false;
       final File crashMarkerFile = new File(options.getCacheDirPath(), NATIVE_CRASH_MARKER_FILE);
+      // TODO: this should probably check whether the Native SDK integration is currently enabled or
+      // remove the marker file if it isn't. Otherwise, application that disable the Native SDK,
+      // will report a crash for the last run forever.
```

Comment:
This is unrelated to this PR, but it can serve as a trigger to fix it: if the native SDK was once enabled and then disabled, do we have a cleanup process elsewhere that would remove the old native marker? If the marker's lifecycle is managed only by the Native SDK, we would report `crashedLastRun == true` forever.